python/src/dataset.rs

```diff
@@ -1860,6 +1861,21 @@ pub fn get_write_params(options: &Bound<'_, PyDict>) -> PyResult<Option<WritePar
             p.enable_v2_manifest_paths = enable_v2_manifest_paths;
         }
 
+        if let Some(auto_cleanup) = get_dict_opt::<Bound<PyDict>>(options, "auto_cleanup_options")?
+        {
+            let interval = get_dict_opt::<String>(&auto_cleanup, "interval")?
+                .and_then(|s| s.parse::<usize>().ok())
+                .unwrap_or(20);
+            let older_than = get_dict_opt::<String>(&auto_cleanup, "older_than_seconds")?
+                .and_then(|s| s.parse::<i64>().ok())
+                .unwrap_or(14 * 24 * 3600); // 14 days
```

Comment:
```suggestion
            let mut auto_cleanup = AutoCleanupParams::default();
            if let Some(interval) = get_dict_opt::<String>(&auto_cleanup, "interval")?
                .and_then(|s| s.parse::<usize>().ok()) {
                    auto_cleanup.interval = interval;
                }
            if let Some(older_than) = get_dict_opt::<String>(&auto_cleanup, "older_than_seconds")?
                .and_then(|s| s.parse::<i64>().ok()) {
                    auto_cleanup.older_than = TimeDelta::seconds(older_than);
                }
```

I think my formatting is bad but hopefully this communicates the idea.  If we can express it in this way then we can avoid repeating the default values which helps keep one source of truth in case we ever need to change the defaults.